mods/ra/maps/production-disruption/production-disruption.lua

```diff
@@ -0,0 +1,270 @@
+RifleSquad2 = { Rifle4, Rifle5, Rifle6 }
+Heavys = { Heavy1, Heavy2 }
+DemoEngiPath = { WaterEntry1.Location, Beach1.Location }
+DemoEngiTeam = { "dtrk", "dtrk", "e6", "e6", "e6" }
+SovietWaterEntry1 = { WaterEntry2.Location, Beach2.Location }
+SovietWaterEntry2 = { WaterEntry2.Location, Beach3.Location }
+SovietSquad = { "e1", "e1", "e1", "e4", "e4" }
+V2Squad = { "v2rl", "v2rl" }
+SubEscapePath = { SubPath1, SubPath2, SubPath3 }
+
+IdleHunt = function(unit) if not unit.IsDead then Trigger.OnIdle(unit, unit.Hunt) end end
+
+MissionStart = function()
+	LZCamera = Actor.Create("camera", true, { Owner = greece, Location = LZ.Location })
+	Chalk1.TargetParatroopers(LZ.CenterPosition, Angle.New(740))
+	if Map.LobbyOption("difficulty") == "normal" then
+		Actor.Create("tsla", true, { Owner = ussr, Location = EasyCamera.Location })
+		Actor.Create("4tnk", true, { Owner = ussr, Facing = 127, Location = Mammoth.Location })
+		Actor.Create("4tnk", true, { Owner = ussr, Facing = 127, Location = Mammoth.Location + CVec.New(1,0) })
+		Actor.Create("v2rl", true, { Owner = ussr, Facing = 127, Location = V2.Location })
+	end
+
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Chalk2.TargetParatroopers(LZ.CenterPosition, Angle.New(780))
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		UnitsArrived = true
+		TeslaCamera = Actor.Create("camera", true, { Owner = greece, Location = TeslaCam.Location })
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(10), function()
+		LZCamera.Destroy()
+		Utils.Do(RifleSquad1, function(actor)
+			if not actor.IsDead then
+				actor.AttackMove(LZ.Location)
+				actor.Hunt()
+			end
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(15), function()
+		TeslaCamera.Destroy()
+		Utils.Do(RifleSquad2, function(actor)
+			if not actor.IsDead then
+				actor.AttackMove(LZ.Location)
+				actor.Hunt()
+			end
+		end)
+	end)
+end
+
+SetupTriggers = function()
+	Trigger.OnDamaged(SubPen, function()
+		Utils.Do(Heavys, function(actor)
+			if not actor.IsDead then
+				actor.Hunt()
+			end
+		end)
+	end)
+
+	Trigger.OnKilled(Church, function()
+		Actor.Create("healcrate", true, { Owner = greece, Location = ChurchCrate.Location })
+	end)
+
+	Trigger.OnKilled(ObjectiveDome, function()
+		if not DomeCaptured == true then
+			greece.MarkFailedObjective(CaptureDome)
+		end
+	end)
+
+	Trigger.OnKilledOrCaptured(SubPen, function()
+		greece.MarkCompletedObjective(StopProduction)
+	end)
+end
+
+PowerDown = false
+PowerDownTeslas = function()
+	if not PowerDown == true then
+		CaptureDome = greece.AddSecondaryObjective("Capture the enemy radar dome.")
+		greece.MarkCompletedObjective(PowerDownTeslaCoils)
+		Media.PlaySpeechNotification(greece, "ReinforcementsArrived")
+		PowerDown = true
+		local demoEngis = Reinforcements.ReinforceWithTransport(greece, LSTType, DemoEngiTeam, DemoEngiPath, { DemoEngiPath[1] })[2][1]
+		Trigger.OnAddedToWorld(demoEngis, function()
+			CaptureRadarDome()
+		end)
```

Comment:
```suggestion
		local demoEngis = Reinforcements.ReinforceWithTransport(greece, LSTType, DemoEngiTeam, DemoEngiPath, { DemoEngiPath[1] })[2]
		Trigger.OnAllRemovedFromWorld(Utils.Where(demoEngis, function(a) return a.Type == "e6"), function()
			if not DomeCaptured then
				greece.MarkFailedObjective(CaptureDome)
			end
		end)

		Trigger.OnCapture(ObjectiveDome, function()
			DomeCaptured = true
			greece.MarkCompletedObjective(CaptureDome)
			SendChronos()
		end)
```
and removing `CaptureRadarDome` should do the trick without timing hacks.